nengo_loihi/utils/sparse_matrix.py

```diff
@@ -0,0 +1,126 @@
+import warnings
```

Comment:
What about putting this in `builder/sparse_matrix` rather than `utils/sparse_matrix`? Just feels like more than "utils" to me, since they're required for the connection building.

Reply:
I would also really want to avoid making a whole new submodule (`utils`) for one thing. My dislike of a `utils` submodule in general is also well documented

Reply:
I put them in `utils` because the `is_` functions are used other places than just the `builder/` folder (namely `block.py`). But I'd be okay with just having them in `builder/` anyway (though I would call the file `sparse_matrix_utils.py` I think, just to make it clear that that file is not for building sparse matrices), or having the file in the main directory.